### PR TITLE
Add assignment 2 files (without tests)

### PR DESCRIPTION
**Pull request by lizaku**

### CODE REVIEW

File: lizaku/assignment2/Q1.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# In my solution I create classes for Node (for storing the key and the information about children;
```
Comment (ariava) [inline lizaku/assignment2/Q1.py:1]: Good job explaining your design choices!

File: lizaku/assignment2/Q1.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    half_data = len(data) // 2
+    left_subtree = data[1:half_data + 1]
+    right_subtree = data[half_data + 1:]
+    root.left = create_tree(left_subtree)
+    root.right = create_tree(right_subtree)
+    return root
+    
+
+def find_ancestors(cur_node, key):
+    # The algorithm is designed as follows: starting from the root, we recursively traverse the tree in pre-order.
```
Comment (ariava) [inline lizaku/assignment2/Q1.py:42]: Awesome explanation!

File: lizaku/assignment2/Q1.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    # When the given key is found, we start an array with ancestors, and on previous steps of the recursion 
+    # the ancestors are collected. If the key is not found, nothing is collected. The key can then be deleted from the 
+    # resulting array. If the key is not present in the tree, returns None.
+    if cur_node.value == key: #found the value
+        return [key]
+    if cur_node.value is None: # did not found the value, reaching the leaves
+        return None
```
Comment (ariava) [inline lizaku/assignment2/Q1.py]: Can you raise a KeyError when you don't find the key in the tree?
In Python, None is a perfectly valid value that could even be contained in a tree. The general rule is to raise an Error when an abnormal condition is detected.

File: lizaku/assignment2/Q2.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
+from Q1 import Node, BinaryTree, create_tree
```
Comment (ariava) [inline lizaku/assignment2/Q2.py:1]: Good job reusing the code!

File: lizaku/assignment2/Q2.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def find_lca(cur_node, node1, node2):
+    # This algorithm is designed as follows: I start with the root and
```
Comment (ariava) [inline lizaku/assignment2/Q2.py]: Great explanation!

File: lizaku/assignment2/Q2.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+def find_lca(cur_node, node1, node2):
+    # This algorithm is designed as follows: I start with the root and 
+    # try to check whether one node is present in the left subtree and other node
+    # is present in the right subtree. If this is the case, then the current node
+    # is LCA. If the traversal reaches the leaves and the value is not found, the recursion step
+    # returns None. If required values are not found in the right subtree,
+    # Then LCA should be found in the left subtree, and I start to inspect it.
+    # Otherwise, I start to inspect the right subtree.
+    if cur_node is None:
+        return None
```
Comment (ariava) [inline lizaku/assignment2/Q2.py]: Can you raise a KeyError when at least one of the values is not found in the tree?

**Review by ariava — CHANGES_REQUESTED** [review]
Good job commenting your algorithms! Please have a look at my review.

**Comment by lizaku** [comment]
Hello @ariava! Thank you for your suggestion about adding exceptions, I forgot about this possibility. Q1 is okay now, and with Q2 I still have problems: when both values are not found, it's alright, but when one value is found, it recursively returns and falls into the condition where the error is not raised. I don't know how to distinguish in this case between cases when one value is found I want to search in this subtree and when one value is found and another value is not found at all (so it's missing).

**Review by ariava — APPROVED** [review]
Awesome! Please feel free to merge this.

**Pull request merged**